interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/test/ModelDataTestInfoProvider.scala

```diff
@@ -25,13 +25,14 @@ class ModelDataTestInfoProvider(modelData: ModelData) extends TestInfoProvider w
   private lazy val nodeCompiler = new NodeCompiler(modelData.modelDefinition, SubprocessComponentDefinitionExtractor(modelData), expressionCompiler, modelData.modelClassLoader.classLoader, ProductionServiceInvocationCollector, ComponentUseCase.TestDataGeneration)
 
   override def getTestingCapabilities(scenario: CanonicalProcess): TestingCapabilities = {
-    collectAllSources(scenario)
-      .map(getTestingCapabilities(_, scenario.metaData))
-      .foldLeft(TestingCapabilities.Disabled)((tc1, tc2) => TestingCapabilities(
-        canBeTested = tc1.canBeTested || tc2.canBeTested,
-        canGenerateTestData = tc1.canGenerateTestData || tc2.canGenerateTestData,
-        canTestWithForm = tc1.canTestWithForm || tc2.canTestWithForm,
-      ))
+    collectAllSources(scenario).map(getTestingCapabilities(_, scenario.metaData)) match {
+      case Nil => TestingCapabilities.Disabled
+      case s => s.reduce((tc1, tc2) => TestingCapabilities (
+          canBeTested = tc1.canBeTested || tc2.canBeTested,
+          canGenerateTestData = tc1.canGenerateTestData || tc2.canGenerateTestData,
+          canTestWithForm = tc1.canTestWithForm && tc2.canTestWithForm,
```

Comment:
Why you changed it? Shouldn't we allow test with form when at least one source supports it?

Reply:
Since _(for now)_ we support test with form for only one source we handle well situations where 
- we have multiple compatible sources (implement TestWithParametersSupport)
- we have multiple incompatible source (do not implement TestWithParametersSupport)

But we do not handle well situations where we have at least one compatible and one incompatible source in one scenario. 
It seemed easiest to not allow in that case to test with form - we can change it when we add support for testing multiple sources - I can add comment for that.

Reply:
ok, let's add info that it is temporary solution and TODO to change it to `or`